examples/tutorials/hybrid_demucs_tutorial.py

```diff
@@ -0,0 +1,346 @@
+# 
+# While doing this, one of the most important steps is to ensure some
+# overlap between each of the chunks, to accomodate for artifacts at the
+# edges. This process of chunking and arrangement can be done in various
+# ways, but an example implementation can be seen in the next few cells.
+# 
+
+from torch.nn import functional as F
+from torchaudio.transforms import Fade
+
+# just leave overlap and segmenent, If device is provided it can be useful in some cases when separating very long tracks, cross fading at the 
+def apply_model(model, mix, segment=10.,
+                overlap=0.1, device=None,
+                ):
+    """
+    Apply model to a given mixture. Use fade, and add segments together in order to add model segment by segment. 
+
+    Args:
+        segment (int): What the segment length will be in seconds
+        device (torch.device, str, or None): if provided, device on which to
+            execute the computation, otherwise `mix.device` is assumed.
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+    """
+    if device is None:
+        device = mix.device
+    else:
+        device = torch.device(device)
+
+    model.to(device)
+    batch, channels, length = mix.shape
+    
+    chunk_len = int(sample_rate * segment * (1 + overlap))
+    prev = 0
+    index = prev + chunk_len
+    fade = Fade(fade_in_len = 0, fade_out_len = int(overlap * sample_rate), fade_shape='linear')
+    
+    final = torch.zeros(batch, 4, channels, length, device=device)
+    
+    print(prev, index)
+    while index < length:
+        chunk = mix[:,:,prev:index]
+        with torch.no_grad():
+            out = model.forward(chunk)
+            
+        if prev != 0:
+            fade = Fade(fade_in_len = int(overlap * sample_rate), fade_out_len = int(overlap * sample_rate), fade_shape='linear')
+        out = fade(out)
+        final[:,:,:,prev:index] += out
+        
+        prev = int(index - overlap * sample_rate)
+        index += chunk_len
+    chunk = mix[:,:,prev:]
+    with torch.no_grad():
+        out = model.forward(chunk)
+    fade = Fade(fade_in_len = int(overlap * sample_rate), fade_out_len = 0, fade_shape='linear')
+    out = fade(out)
+    final[:,:,:,prev:] += out
+    return final
+
+def plot_spectrogram(stft, title="Spectrogram"):
+    magnitude = stft.abs()
+    spectrogram = 20 * torch.log10(magnitude + 1e-8).numpy()
+    figure, axis = plt.subplots(1, 1)
+    img = axis.imshow(spectrogram, cmap="viridis", vmin=-100, vmax=0, origin="lower", aspect="auto")
+    figure.suptitle(title)
+    plt.colorbar(img, ax=axis)
+    plt.show()
+
+
+######################################################################
+# 5. Run Model
+# ------------
+# 
+# Finally, we run the model and store the separate source files in a
+# directory
+# 
+# As a test song, we will be using Strand Of Oaks by Spacestation from
```

Comment:
Could you add a link to the wav file here, or explain where the song file came from, so users can find the file themselves to use and replicate